pkg/deploy/spindeploy/transformer/telemety.go

```diff
@@ -0,0 +1,114 @@
+package transformer
+
+import (
+	"context"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
+	"github.com/armory/spinnaker-operator/pkg/generated"
+	"github.com/armory/spinnaker-operator/pkg/inspect"
+	"github.com/armory/spinnaker-operator/version"
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	EchoConfigFile     = "echo.yml"
+	KubernetesOperator = "kubernetes_operator"
+	TelemetryKey       = "telemetry"
+)
+
+type telemetryTransformer struct {
+	log logr.Logger
+}
+
+type telemetryTransformerGenerator struct{}
+
+type DeploymentMethod struct {
+	DeploymentType    string `json:"type"`
+	DeploymentVersion string `json:"version"`
+}
+
+type Telemetry struct {
+	Enabled                 bool             `json:"enabled"`
+	Endpoint                string           `json:"endpoint"`
+	InstanceId              string           `json:"instanceId"`
+	SpinnakerVersion        string           `json:"spinnakerVersion"`
+	DeploymentMethod        DeploymentMethod `json:"deploymentMethod"`
+	ConnectionTimeoutMillis int              `json:"connectionTimeoutMillis"`
+	ReadTimeoutMillis       int              `json:"readTimeoutMillis"`
```

Comment:
I think we can update `DeploymentMethod` only without having to declare all available fields inside `Telemetry`. The problem is that we would need to change the operator project every time a new field is added to telemetry, which can be avoided.

Reply:
I thought the same at the beginning, I just add them thinking about having all the fields available for future changes.